frontend/public/components/monitoring/alert-manager-receiver-forms.tsx

```diff
@@ -200,143 +210,133 @@ const RoutingLabels = ({ routeLabels, setRouteLabels, saveAsDefaultReceiver }) =
   );
 };
 
-const ReceiverBaseForm: React.FC<ReceiverBaseFormProps> = ({ obj, titleVerb, saveButtonText }) => {
+const ReceiverBaseForm: React.FC<ReceiverBaseFormProps> = ({
+  obj,
+  titleVerb,
+  saveButtonText,
+  editReceiverNamed,
+}) => {
   const secret: K8sResourceKind = obj; // Secret "alertmanager-main" which contains alertmanager.yaml config
   const [errorMsg, setErrorMsg] = React.useState('');
   const [inProgress, setInProgress] = React.useState(false);
   let config: AlertManagerConfig;
   if (!errorMsg) {
     config = getAlertManagerConfig(secret, setErrorMsg);
   }
-  const [receiverName, setReceiverName] = React.useState('');
-  const [receiverType, setReceiverType] = React.useState('');
-  const [webhookUrl, setWebhookUrl] = React.useState('');
-  const defaultPagerDutyValues = {
-    integrationType: 'events',
+
+  let initialReceiverName = '';
+  let initialReceiverType = '';
+  let initialPagerDutyValues = {
+    integrationType: 'events', // 'Events API v2'
     integrationKey: '',
   };
-  const [pagerDutyValues, setPagerDutyValues] = React.useState(defaultPagerDutyValues);
+  let initialWebhookUrl = '';
```

Comment:
This would be a bit easier to reason about if some or all of the `let`s were converted to `const`s.